domain/src/test/java/org/oppia/android/domain/profile/ProfileManagementControllerTest.kt

```diff
@@ -1168,6 +1171,83 @@ class ProfileManagementControllerTest {
     assertThat(lastShownTimeMs).isEqualTo(DEFAULT_SURVEY_LAST_SHOWN_TIMESTAMP_MILLIS)
   }
 
+  @Test
+  fun testFetchLastSelectedClassroomId_updateClassroomId_checkUpdateIsSuccessful() {
+    setUpTestApplicationComponent()
+    addTestProfiles()
+
+    monitorFactory.ensureDataProviderExecutes(
+      profileManagementController.loginToProfile(PROFILE_ID_0)
+    )
+
+    monitorFactory.waitForNextSuccessfulResult(
+      profileManagementController.updateLastSelectedClassroomId(
+        PROFILE_ID_0,
+        TEST_CLASSROOM_ID_1
+      )
+    )
+
+    val lastSelectedClassroomId = monitorFactory.waitForNextSuccessfulResult(
+      profileManagementController.retrieveLastSelectedClassroomId(PROFILE_ID_0)
+    )
+
+    assertThat(lastSelectedClassroomId).isEqualTo(TEST_CLASSROOM_ID_1)
+  }
+
+  @Test
+  fun testFetchLastSelectedClassroomId_updateClassroomIds_checkUpdateIsSuccessfulPerProfile() {
+    setUpTestApplicationComponent()
+    addTestProfiles()
+
+    // Login to profile 0 and update the last selected classroom to classroom 1.
+    monitorFactory.ensureDataProviderExecutes(
+      profileManagementController.loginToProfile(PROFILE_ID_0)
+    )
+    monitorFactory.waitForNextSuccessfulResult(
+      profileManagementController.updateLastSelectedClassroomId(
+        PROFILE_ID_0,
+        TEST_CLASSROOM_ID_1
+      )
+    )
+
+    // Login to profile 1 and update the last selected classroom to classroom 2.
+    monitorFactory.ensureDataProviderExecutes(
+      profileManagementController.loginToProfile(PROFILE_ID_1)
+    )
+    monitorFactory.waitForNextSuccessfulResult(
+      profileManagementController.updateLastSelectedClassroomId(
+        PROFILE_ID_1,
+        TEST_CLASSROOM_ID_2
+      )
+    )
+
+    // Verify that last selected classroom of profile 0 is classroom 1.
+    val classroomIdProfile0 = monitorFactory.waitForNextSuccessfulResult(
```

Comment:
Rename to be more descriptive.
```suggestion
    val profile0SelectedClassroomId = monitorFactory.waitForNextSuccessfulResult(
```

Reply:
Done.